src/Common/src/System/Net/Logging/HttpEventSource.cs

```diff
@@ -0,0 +1,161 @@
+    {
+        private const int ASSOCIATE_ID = 1;
+        private const int URI_BASE_ADDRESS_ID = 2;
+        private const int CONTENT_NULL_ID = 3;
+        private const int CLIENT_SEND_COMPLETED = 4;
+        private const int HEADERS_INVALID_VALUE_ID = 5;
+
+        private readonly static HttpEventSource s_log = new HttpEventSource();
+        private HttpEventSource() { }
+        public static HttpEventSource Log
+        {
+            get
+            {
+                return s_log;
+            }
+        }
+
+        internal static void Associate(object objA, object objB)
+        {
+            if (!s_log.IsEnabled())
+            {
+                return;
+            }
+            s_log.Associate(LoggingHash.GetObjectName(objA),
+                            LoggingHash.HashInt(objA),
+                            LoggingHash.GetObjectName(objB),
+                            LoggingHash.HashInt(objB));
+        }
+
+        [Event(ASSOCIATE_ID, Keywords = Keywords.Default,
+            Level = EventLevel.Informational, Message = "[{0}#{1}]<-->[{2}#{3}]")]
+        internal unsafe void Associate(string objectA, int objectAHash, string objectB, int objectBHash)
+        {
+            const int SIZEDATA = 4;
+            fixed (char* arg1Ptr = objectA, arg2Ptr = objectB)
+            {
+                EventData* dataDesc = stackalloc EventSource.EventData[SIZEDATA];
+
+                dataDesc[0].DataPointer = (IntPtr)(arg1Ptr);
+                dataDesc[0].Size = (objectA.Length + 1) * sizeof(char); // Size in bytes, including a null terminator. 
+                dataDesc[1].DataPointer = (IntPtr)(&objectAHash);
+                dataDesc[1].Size = sizeof(int);
+                dataDesc[2].DataPointer = (IntPtr)(arg2Ptr);
+                dataDesc[2].Size = (objectB.Length + 1) * sizeof(char);
+                dataDesc[3].DataPointer = (IntPtr)(&objectBHash);
+                dataDesc[3].Size = sizeof(int);
+
+                WriteEventCore(ASSOCIATE_ID, SIZEDATA, dataDesc);
+            }
+        }
+
+        [NonEvent]
+        internal static void UriBaseAddress(object obj, string baseAddress)
+        {
+            if (!s_log.IsEnabled())
+            {
+                return;
+            }
+            s_log.UriBaseAddress(baseAddress, LoggingHash.GetObjectName(obj), LoggingHash.HashInt(obj));
+        }
+
+        [Event(URI_BASE_ADDRESS_ID, Keywords = Keywords.Debug,
+            Level = EventLevel.Informational)]
+        internal unsafe void UriBaseAddress(string uriBaseAddress, string objName, int objHash)
+        {
+            const int SIZEDATA = 3;
+            fixed (char* arg1Ptr = uriBaseAddress, arg2Ptr = objName)
+            {
+                EventData* dataDesc = stackalloc EventSource.EventData[SIZEDATA];
+
+                dataDesc[0].DataPointer = (IntPtr)(arg1Ptr);
+                dataDesc[0].Size = (uriBaseAddress.Length + 1) * sizeof(char); // Size in bytes, including a null terminator. 
```

Comment:
No, but this is the pattern suggested in the eventsource guide, and is also the way the predefined WriteCore methods work. 
https://github.com/dotnet/coreclr/blob/ed1867bc8e445806ebe2e05a5d3515787006321d/src/mscorlib/src/System/Diagnostics/Eventing/EventSource.cs